src/library_webgl.js

```diff
@@ -82,10 +82,24 @@ var LibraryGL = {
     return HEAPU16;
   },
 
+  // Returns the appropriate amount of shifts for a heap (based on the element
+  // size in the heap).
   $heapAccessShiftForWebGLHeap: function(heap) {
     return 31 - Math.clz32(heap.BYTES_PER_ELEMENT);
   },
 
+  // Receives a pointer and a heap, and shifts the pointer the appropriate
+  // amount of times for that heap (based on the element size in the heap).
+  $getShiftedPtrForWebGLHeap__deps: ['$heapAccessShiftForWebGLHeap'],
+  $getShiftedPtrForWebGLHeap: function(ptr, heap) {
+    var shifts = heapAccessShiftForWebGLHeap(heap);
+#if CAN_ADDRESS_2GB
+    return ptr >>> shifts;
+#else
+    return ptr >> shifts;
+#endif
+  },
```

Comment:
In general I think this would be nice to do compile-time, #15433 can help with that. This way there wouldn't be a WebGL-specific function emitted to grow code size. But lgtm for now.